lib/src/main/java/com/knuddels/jtokkit/EncodingFactory.java

```diff
@@ -7,162 +7,169 @@
-		try (final InputStream in = EncodingFactory.class.getResourceAsStream(fileName)) {
-			if (in == null) {
-				throw new IllegalStateException("Could not find " + fileName + " in resources");
-			}
-
-			final Map<byte[], Integer> mergeableRanks = new HashMap<>();
-			final BufferedReader reader = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
-			String line;
-			while ((line = reader.readLine()) != null) {
-				final String[] parts = line.split("\\s+", 2);
-				if (parts.length != 2) {
-					throw new IllegalStateException("Invalid line in " + fileName + ": " + line);
-				}
-
-				final byte[] token = Base64.getDecoder().decode(parts[0].getBytes(StandardCharsets.UTF_8));
-				final int rank = Integer.parseInt(parts[1]);
-
-				mergeableRanks.put(token, rank);
-			}
-
-			return mergeableRanks;
-		} catch (final IOException e) {
-			throw new IllegalStateException("Could not load " + fileName + " from resources", e);
-		}
-	}
-
-	private EncodingFactory() {
-	}
+import static java.lang.Integer.parseInt;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class EncodingFactory {
+    private static final Map<String, Integer> SPECIAL_TOKENS_CL100K_BASE;
+    private static final Map<String, Integer> SPECIAL_TOKENS_X50K_BASE;
+    private static final Map<String, Integer> SPECIAL_TOKENS_P50K_EDIT;
+
+    private static final String ENDOFTEXT = "<|endoftext|>";
+    private static final String FIM_PREFIX = "<|fim_prefix|>";
+    private static final String FIM_MIDDLE = "<|fim_middle|>";
+    private static final String FIM_SUFFIX = "<|fim_suffix|>";
+    private static final String ENDOFPROMPT = "<|endofprompt|>";
+
+    static {
+        Map<String, Integer> map = new HashMap<>();
+        map.put(ENDOFTEXT, 50256);
+        SPECIAL_TOKENS_X50K_BASE = Collections.unmodifiableMap(map);
+    }
+
+    static {
+        Map<String, Integer> map = new HashMap<>();
+        map.put(ENDOFTEXT, 50256);
+        map.put(FIM_PREFIX, 50281);
+        map.put(FIM_MIDDLE, 50282);
+        map.put(FIM_SUFFIX, 50283);
+        SPECIAL_TOKENS_P50K_EDIT = Collections.unmodifiableMap(map);
+    }
+
+    static {
+        Map<String, Integer> map = new HashMap<>();
+        map.put(ENDOFTEXT, 100257);
+        map.put(FIM_PREFIX, 100258);
+        map.put(FIM_MIDDLE, 100259);
+        map.put(FIM_SUFFIX, 100260);
+        map.put(ENDOFPROMPT, 100276);
+        SPECIAL_TOKENS_CL100K_BASE = Collections.unmodifiableMap(map);
+    }
+
+    private EncodingFactory() {
+    }
+
+    /**
+     * Returns an {@link Encoding} instance for the r50k_base encoding.
+     *
+     * @return an {@link Encoding} instance for the r50k_base encoding
+     */
+    public static Encoding r50kBase() {
+        return fromPredefinedParameters(
+                "r50k_base",
+                "'(?:[sdmt]|ll|ve|re)| ?\\p{L}+| ?\\p{N}+| ?[^\\s\\p{L}\\p{N}]+|\\s+(?!\\S)|\\s+",
```

Comment:
this is the meat of the change: by prefixing all contraction by a single `'`, the engine can rule out all combinations.
I've also rearranged the letters by how likely they're encountered in English.
The only non-english words these will match is `'ſ` (https://en.wikipedia.org/wiki/Long_s), but that's most likely not intended by the authors. This change will still match that corner-case.